micrometer-observation/src/main/java/io/micrometer/observation/aop/ObservedAspect.java

```diff
@@ -142,7 +140,7 @@ public Object observeMethod(ProceedingJoinPoint pjp) throws Throwable {
         return observe(pjp, method, observed);
     }
 
-    private Object observe(ProceedingJoinPoint pjp, Method method, Observed observed) throws Throwable {
+    private @Nullable Object observe(ProceedingJoinPoint pjp, Method method, Observed observed) throws Throwable {
```

Comment:
I suppose we don't know the nullability of the instrumented method's return value. Should we make it `@NullUnmarked`?

Reply:
Semantically, the difference would be this?
`@Nullable`: we know that it can return `null`.
`@NullUnmarked`: we have no idea (it can but there will be no warning)

Let's discuss this, I feel we have a lot of similar use cases (`Observation.observe`, `Timer.record`, etc. where we accept something from the user we call and return its value and it's up to the user if that something can return `null` or not).

Reply:
I'm willing to be convinced otherwise, but semantically it feels like using `@NullUnmarked` on the return type is the most accurate. I can imagine a user complaining if we mark it as `@Nullable` but they know in their usage it isn't nullable because the Callable/Supplier/Function they pass does not have a nullable return value.